README.md

```diff
@@ -300,13 +300,19 @@ The *AzureIoTHub.Message* class is used to create an event object to send to IoT
 
 ### AzureIoTHub.Message Class Usage ###
 
-#### Constructor: AzureIoTHub.Message(*message[, properties]*) ####
+#### Constructor: AzureIoTHub.Message(*message[, properties][, encoder]*) ####
 
-The constructor takes one required parameter, *message*, which can be created from a string or any object that can be converted to JSON. It may also take an optional parameter: a table of message properties.
+The constructor takes one required parameter, *message*, which can be created from a string or any object that can be converted to JSON. 
+
+There are also two optional parameters in the following order:
+
+- properties: a table of message properties
+- encoder: a custom JSON encoder function for encoding the provided data (e.g. [JSONEncoder.encode](https://github.com/electricimp/JSONEncoder))
 
 ```squirrel
 local message1 = AzureIoTHub.Message("This is an event");
 local message2 = AzureIoTHub.Message({ "id": 1, "text": "Hello, world." });
+local message3 = AzureIoTHub.Message({ "a": JSONLiteralString("1234567890") }, null, JSONEncoder.encode.bindenv(JSONEncoder))
```

Comment:
can you please either define `JSONLiteralString` or provide the raw string here. otherwise, this might be confusing.

Reply:
Just updated to include `JSONLiteralString` definition in the example. I thought that was actually a library you guys were maintaining but it's not (I think it's something @deldrid1 threw together).

It's very simply -- maybe something to maintain in the electricimp set of repos since JSONEncoder has native support for it